search/find-place/src/test/java/com/esri/samples/find_place/FindPlaceTest.java

```diff
@@ -0,0 +1,206 @@
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.testfx.api.FxRobot;
+import org.testfx.api.FxToolkit;
+import org.testfx.framework.junit.ApplicationTest;
+import org.testfx.robot.Motion;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+public class FindPlaceTest extends FxRobot {
+
+  public static final int SLEEP_500MS = 500;
+  public static final int SLEEP_1000MS = 1000;
+  public static final int SLEEP_1500MS = 1500;
+
+  MapView mapView;
+  GraphicsOverlay graphicsOverlay;
+  ComboBox<String> placeBox;
+  ComboBox<String> locationBox;
+  Node locationBoxArrowRegion;
+  Node placeBoxArrowRegion;
+  Button searchButton;
+
+  @Before
+  public void setup() throws Exception {
+    // launch the sample
+    ApplicationTest.launch(FindPlaceSample.class);
+
+    // get a handle on the MapView and GraphicsOverlay
+    Node mapViewNode = lookup("#mapView").query();
+    if (mapViewNode instanceof MapView) {
+      mapView = (MapView) mapViewNode;
+      graphicsOverlay = mapView.getGraphicsOverlays().get(0);
+
+    } else {
+      fail("MapView Node could not be found");
+    }
+
+    // get a handle on the ComboBoxes
+    placeBox = lookup("#placeBox").queryComboBox();
+    locationBox = lookup("#locationBox").queryComboBox();
+
+    // get a handle on the ComboBoxes' arrow buttons
+    for (Node child : locationBox.getChildrenUnmodifiable()) {
+      if (child.getStyleClass().contains("arrow-button")) {
+        locationBoxArrowRegion = ((Pane) child).getChildren().get(0);
+      }
+    }
+
+    for (Node child : placeBox.getChildrenUnmodifiable()) {
+      if (child.getStyleClass().contains("arrow-button")) {
+        placeBoxArrowRegion = ((Pane) child).getChildren().get(0);
+      }
+    }
+
+    // get a handle on the search button
+    searchButton = lookup("#searchButton").queryButton();
+
+    // wait for initialization
+    sleep(SLEEP_1000MS);
+  }
+
+  @After
+  public void tearDown() throws Exception {
+    FxToolkit.cleanupStages();
+    // wait before the next test
+    sleep(SLEEP_1000MS);
+  }
+
+  /**
+   * Test Case 1: When using one of the pre-defined search inputs, the app returns the expected number of results
+   */
+  @Test
+  public void defaultSearchInput() {
+
+    clickOn(placeBoxArrowRegion);
+    clickOn("Starbucks");
+
+    clickOn(locationBoxArrowRegion);
+    clickOn("Los Angeles, CA");
+
+    clickOn(searchButton).sleep(SLEEP_1500MS);
+
+    // assert that the expected number of graphics (location pins) is displayed
+    assertEquals("Unexpected number of graphics (location pins) found", 50, graphicsOverlay.getGraphics().size());
+  }
+
+  /**
+   * Test Case 2: When using a custom search query, the app returns the expected number of results
+   */
+  @Test
+  public void customSearchInput() {
```

Comment:
I have noticed a bug in this sample where if you clear the place input (or edit it significantly), the default options go missing. Would be good to have a test for that and use it to fix the bug and verify.